pkg/apis/apps/types.go

```diff
@@ -101,10 +101,10 @@ type RollingUpdateStatefulSetStrategy struct {
 	// The maximum number of pods that can be unavailable during the update.
 	// Value can be an absolute number (ex: 5) or a percentage of desired pods (ex: 10%).
 	// Absolute number is calculated from percentage by rounding up. This can not be 0.
-	// Defaults to 1. This field is alpha-level and is only honored by servers that enable the
-	// MaxUnavailableStatefulSet feature. The field applies to all pods in the range 0 to
+	// Defaults to 1. This field is beta-level and is enabled by default. The field applies to all pods in the range 0 to
 	// Replicas-1. That means if there is any unavailable pod in the range 0 to Replicas-1, it
 	// will be counted towards MaxUnavailable.
+	// +featureGate=MaxUnavailableStatefulSet
```

Comment:
As you wrote in [beta requirements](https://github.com/kubernetes/enhancements/blob/master/keps/sig-apps/961-maxunavailable-for-statefulset/README.md#beta) can you add a note here, that this applies to `ParallelPodManagement` because for for `OrderedReadyPodManagement` we will continue rolling one at a time, so this field will not be effective. 

Additionally, I asked api-reviewer for opinions about potentially introducing stricter validation for this field. See https://kubernetes.slack.com/archives/CJUQN3E4T/p1758188879687169. At minimum you'll need to follow https://github.com/kubernetes/community/blob/master/contributors/devel/sig-architecture/api_changes.md#ratcheting-validation (there's a great example how to do that).

Reply:
Just talked with Jordan about it, and we both agree that because we already have several `MaxUnavailable` instances which are not validated to be bigger than `.spec.replicas` (for DaemonSet, Deployments) so it's also not required for having that verification in this case. So **no validation** will be added here.